src/gluonnlp/models/transformer.py

```diff
@@ -0,0 +1,1247 @@
+import numpy as np
+import mxnet as mx
+from mxnet import use_np
+from mxnet.gluon import nn, Block, HybridBlock
+from typing import Optional, Tuple, List
+from ..utils.registry import Registry
+from ..attention_cell import MultiHeadAttentionCell, gen_self_attn_mask, gen_mem_attn_mask
+from ..layers import PositionalEmbedding, PositionwiseFFN, InitializerType
+from ..utils.config import CfgNode as CN
+from ..sequence_sampler import SequenceDecoder
+__all__ = ['TransformerEncoderLayer', 'TransformerDecoderLayer',
+           'TransformerEncoder', 'TransformerDecoder',
+           'TransformerNMTModel', 'TransformerNMTInference']
+
+transformer_nmt_cfg_reg = Registry('transformer_nmt_cfg')
```

Comment:
@szha Another example, in which we created the registry to register configurations.